### DISCUSSION TITLE
Make Control Nodes be able to play audios by themself.

### DISCUSSION THREAD

**Discussion by PPBoxHead** [godotengine/godot-proposals#7999] [GUI]
The idea is quite simple. Yet it can add a bit of complexity to the already complex but powerful GUI workflow of Godot, the idea is add a group tab in Control class or in a Theme resource (can be another good option) to add Stream (audio) resources to play depending on the interaction the user has with the GUI. 
On the same way we can change text and styleboxes on buttons and so, add a way to play sounds without needing of make a SFXs manager or add a lot of AudioStreamPlayers for each action or signal emitted by controls depending on the interaction, but this proposal is thinked to be implemented on GUIs that are more heavy or complex (also, would be perfect to don't need to mess around with all the code thing makind menues). Might can be done with the AudioServer or in help with that feature. 
Just thought about it today and wanted to write a proposal on the theme. 
I apologize for my English skills writing so long texts.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/1472.
